src/viking.js

```diff
@@ -1,11 +1,91 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack() {
+    return this.strength;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`;
+    } else {
+      return `${this.name} has died in act of combat`;
+    }
+  }
+  battleCry() {
+    return `Odin Owns You All!`;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(viking) {
+    this.vikingArmy.push(viking);
+  }
+  addSaxon(saxon) {
+    this.saxonArmy.push(saxon);
+  }
+  vikingAttack() {
+    let randomViking =
+      this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+    let randomSaxon =
+      this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    let damage = randomViking.strength;
+    let fight = randomSaxon.receiveDamage(damage);
+    if (randomSaxon.health <= 0) {
+      this.saxonArmy.pop(randomSaxon);
```

Comment:
Miguel, here, using the `.pop()` method to remove your dead saxon from the array, always removes the last index of the array even if you pass a determined index to it. In this case, Jasmine is accepting that answer, but the most correct way to do that would be using the `.splice()` method, where you can define an index to start removing and the amount that you want to remove from the array.
So, it would be something like this:
```
if (randomSaxon.health <= 0) {
      this.saxonArmy.splice(randomSaxon, 1);
    }
```